internal/common/resource.go

```diff
@@ -9,29 +9,64 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
-type ResourceUpdateFunc = func(expected, found controllerutil.Object)
+type StatusMessage = *string
 
-func CreateOrUpdateResource(request *Request, resource controllerutil.Object, updateResource ResourceUpdateFunc) error {
-	err := controllerutil.SetControllerReference(request.Instance, resource, request.Scheme)
-	if err != nil {
-		return err
+type ResourceStatus struct {
+	Resource     controllerutil.Object
+	Progressing  StatusMessage
+	NotAvailable StatusMessage
+	Degraded     StatusMessage
+}
+
+func ReconcileMultiple(reqest *Request, funcs ...func(*Request) (ResourceStatus, error)) ([]ResourceStatus, error) {
```

Comment:
I think renaming this to `Reconcile()` with `...func()` as an argument is more readable

Reply:
I have renamed it to `CollectResourceStatus()`, because the function does not do any reconciliation.